keps/sig-node/2535-ensure-secret-pulled-images/README.md

```diff
@@ -738,13 +736,31 @@ the behavior of the pull policies will revert to the previous behavior.
 
 ###### What specific metrics should inform a rollback?
 
-If the feature gate is enabled, the kubelet will gather metrics `image_pull_secret_recheck_miss` and
-`image_pull_secret_recheck_hit` which are both histograms counting the number of images that had a cache miss/hit.
-
-This will allow an admin to see how many images have authorization checks done.
-
-A histogram was chosen to allow an admin to compare registry uptime with cache misses, as the main failure scenerio is registry unavailability
-could cause pods not to come up, because the kubelet doesn't have credentials cached.
+Enabling the feature will make the kubelet expose several metrics:
+- each caching layers should provide the following:
+    - `<cache_name>_<pullintents/pulledrecords>_total` gauge for the number of records (e.g. files, keys of a map)
+      that are kept in the cache
+- `image_mustpull_checks_total` counter vector tracks how many times the check for
+  image pull credentials was called
+  - labels:
+    - `result`:
+      - "policyAllowed" for cases where the kubelet's pull policy allows
+        access to the image (e.g. via allowlist or to an image pre-pulled to the node)
+      - "recordFound" when a matching record was found in the cache
+      - "mustRepull" when a new pull is required to access the image
+      - "error" in error cases
+- the `image_ensure_requests_total` counter vector tracks how many times EnsureImageExists()
+  was called
+  - labels:
+    - `pull_policy` - either of "never", "ifnotpresent" or "always", according to container's pull policy
+    - `present_locally` - "true" if the image is present on the node, "false" if it isn't,
+                          "unknown" if there was an error before this was determined
+    - `pull_required` - "true" if an image was requested to be pulled (e.g. from credential reverification
```

Comment:
```suggestion
    - `access_authentication_required` - "true" if an image was requested to be pulled (e.g. from credential reverification
```

Reply:
This metric is not related to the credential verification check, but is run from the EnsureImageExists() function.
`pull_required` makes more sense here - it should automatically be true for images with the `Always` pull policy - these won't even go through the creds verification flow, and so we need to be a bit more generic.